pkg/testrunner/result/summary-poster.go

```diff
@@ -0,0 +1,200 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package result
+
+import (
+	"fmt"
+	"github.com/Masterminds/semver"
+	argov1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/gardener/test-infra/pkg/testmachinery/metadata"
+	"github.com/gardener/test-infra/pkg/testrunner"
+	"github.com/gardener/test-infra/pkg/util/slack"
+	"github.com/go-logr/logr"
+	"github.com/olekukonko/tablewriter"
+	"sort"
+	"strings"
+)
+
+var SucessSymbols = map[bool]string{
+	true:  "✅",
+	false: "❌",
+}
+
+const NA = "N/A"
+
+func (c *Collector) postTestrunsSummaryInSlack(config Config, log logr.Logger, runs testrunner.RunList) {
+	if !config.PostSummaryInSlack {
+		return
+	}
+	table, err := renderTableOfRuns(log, runs)
+	if err != nil {
+		log.Error(err, "failed creating a table to post")
+	}
+	if table == "" {
+		log.Info("no table to render")
+		return
+	}
+
+	slackClient, err := slack.New(log, config.SlackToken)
+	if err != nil {
+		log.Error(err, "Was not able to create slack client")
+	}
+
+	concourseURLFooter := ""
+	if config.ConcourseURL != "" {
+		concourseURLFooter = fmt.Sprintf("\nConcourse Job: %s", config.ConcourseURL)
+	}
+
+	if err := slackClient.PostMessage(config.SlackChannel, fmt.Sprintf("```%s\n%s\n%s```%s", header(), table, legend(), concourseURLFooter)); err != nil {
+		log.Error(err, "failed to post the slack message of test summary")
+	}
+}
+
+func header() string {
+	return "Integration Test Results:"
+}
+
+func legend() string {
+	return fmt.Sprintf(`
+%s: Tests succeeded | %s: Tests failed | %s: Tests not applicable
+`, SucessSymbols[true], SucessSymbols[false], NA)
+}
+func renderTableOfRuns(log logr.Logger, runs testrunner.RunList) (string, error) {
+	writer := &strings.Builder{}
+	table := tablewriter.NewWriter(writer)
+	headerKeys := make(map[string]int, 0) // maps the header values to their index
+	header := []string{""}
+
+	for _, run := range runs {
+		meta := run.Metadata
+		if meta.CloudProvider != "" {
+			header = append(header, meta.CloudProvider)
+			headerKeys[meta.CloudProvider] = len(header) - 1
+		}
+	}
+
+	res := results{
+		header:  headerKeys,
+		content: make(map[string]resultRow),
+	}
+
+	for _, run := range runs {
+		meta := run.Metadata
+		if meta.CloudProvider == "" {
+			log.V(5).Info("skipped testrun", "id", meta.Testrun.ID)
+			continue
+		}
+
+		dimensionKey := fmt.Sprintf("%s %s", meta.KubernetesVersion, meta.OperatingSystem)
+		if meta.FlavorDescription != "" {
+			dimensionKey = fmt.Sprintf("%s (%s)", dimensionKey, meta.FlavorDescription)
+		}
+		res.AddResult(meta, run.Testrun.Status.Phase == argov1.NodeSucceeded)
+	}
+	if res.Len() == 0 {
+		return "", nil
+	}
+
+	table.SetHeader(header)
+	table.AppendBulk(res.GetContent())
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.Render()
+	return writer.String(), nil
+}
+
+type resultRow struct {
```

Comment:
looks like all the sorting and maybe some parts of the rendering could be abtracted and shared with the functions from https://github.com/gardener/test-infra/blob/9aad6bdac94870dff02aefa692881d777a02b550/cmd/testrunner/cmd/notify/tablecontent.go#L29

I think there could be a common renderner in `utils` that takes an abstracted model to generate the table something like an inner join of `testrunner.Run` and `result.AssetOverviewItems`? wdyt?

Reply:
can you check again?